crates/starfish/core/src/dag_state.rs

```diff
@@ -1206,7 +1236,7 @@ impl DagState {
 
     #[cfg(test)]
     pub(crate) fn set_pending_acknowledgments(&mut self, acknowledgments: Vec<BlockRef>) {
-        self.pending_acknowledgments = acknowledgments;
+        self.pending_acknowledgments = acknowledgments.into_iter().collect();
```

Comment:
This is more direct and more readable and also moves ownership of the elements.
```suggestion
        self.pending_acknowledgments = BTreeSet::from(acknowledgments);
```